core/src/avm2/object.rs

```diff
@@ -89,12 +94,6 @@ pub trait TObject<'gc>: 'gc + Collect + Debug + Into<Object<'gc>> + Clone + Copy
         name: &QName<'gc>,
         activation: &mut Activation<'_, 'gc, '_>,
     ) -> Result<Value<'gc>, Error> {
-        if !self.has_instantiated_property(name) {
```

Comment:
now that this is removed, `&mut self` is no longer needed, right? Then `mut` removal can propagate into a lot of places, like `resolve()`, and then every place that does `scope.write(gc).resolve(...)` -> `scope.read().resolve(...)` etc.